src/util/number.ts

```diff
@@ -138,7 +138,9 @@ export function round(x: number | string, precision: number, returnStr: false):
 export function round(x: number | string, precision: number, returnStr: true): string;
 export function round(x: number | string, precision?: number, returnStr?: boolean): string | number {
     if (precision == null) {
-        precision = 10;
+        // auto adjust round precision by decimal place length only when the number is using scientific
+        const exp = String(x).split('e')[1];
```

Comment:
It's not the precision `x`. `1.3e-10` will have exp 10 here. But it should be 11. 
Also we prefer x + '' to convert a value to string. And `-exp` already convert it to a number.